Geometry.cs

```diff
@@ -130,14 +130,46 @@ class Polygon {
 class Drawing {
    public void Add (Polygon poly) {
       mPolys.Add (poly);
-      mBound = new (); 
+      mBound = new ();
    }
 
    public IReadOnlyList<Polygon> Polys => mPolys;
-   List<Polygon> mPolys = new ();
+   readonly List<Polygon> mPolys = new ();
+
+   public IReadOnlyList<Point2> ConvexHull {
+      get {
+         // Return the hull if it has been computed for all the polygons.
+         if (mcHull == mPolys.Count) return mHull;
+         // Incrementaly compute the convex hull for the remaining polygons.
+         var pts = mHull.Concat (mPolys.Skip (mcHull).SelectMany (poly => poly.Pts)).ToArray ();
+         // The lowest anchor point.
+         Point2 ptA = pts.MinBy (p => p.Y);
+         pts = pts.OrderBy (p => Normalized (ptA.AngleTo (p))).ToArray ();

```

Comment:
Do we need the Normalized? It seems to me the normal domain of -PI/2 .. +PI/2 should work fine in this case?

Reply:
The Atan2 domain is **-PI .. +PI** (that is how it is able to distinguish between (1, 1) and (-1, -1) slopes). As we already anchor to the lowest point, the new domain should become **0 .. +PI**. **Leaf-fill.txt** has a single point at the lowest Y, but there can be a situation when there are multiple points (from different polygons perhaps) at the **Ymin**. And a round-off at the epsilon boundary may return -PI for some points lying left while +PI for a few (also on the left) to anchor on the same horizontal line. I thought normalizing will make sure that all the points lying left to the anchor on a horizontal line will snap the slope it to +PI, ensuring the **0 .. +PI** domain. 

Or will the correct thing to do is to make sure that the anchor point is also the _left-most_ point at the **Ymin**?

Reply:
_Left-most point at the YMin_ is what the ConvexHull routine in Flux.Geo seems to be doing.

Reply:
OK. Applied this change. Handling the collinear situation better now.